sdk_extensions/aws_v1_support/src/main/java/io/opentelemetry/sdk/extensions/trace/aws/resource/EcsResource.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.extensions.trace.aws.resource;
+
+import static io.opentelemetry.common.AttributeValue.stringAttributeValue;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
+import io.opentelemetry.common.AttributeValue;
+import io.opentelemetry.sdk.resources.ResourceConstants;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+class EcsResource extends AwsResource {
+
+  private static final Logger logger = Logger.getLogger(EcsResource.class.getName());
+
+  private static final String ECS_METADATA_KEY_V4 = "ECS_CONTAINER_METADATA_URI_V4";
+  private static final String ECS_METADATA_KEY_V3 = "ECS_CONTAINER_METADATA_URI";
+  private static final String CONTAINER_ID = "container.id";
+
+  private final DockerHelper dockerHelper;
+  private final String uriV3;
+  private final String uriV4;
+
+  EcsResource() {
+    this(
+        new DockerHelper(), System.getenv(ECS_METADATA_KEY_V3), System.getenv(ECS_METADATA_KEY_V4));
+  }
+
+  @VisibleForTesting
+  EcsResource(DockerHelper dockerHelper, String uriV3, String uriV4) {
+    this.dockerHelper = dockerHelper;
+    this.uriV3 = uriV3;
+    this.uriV4 = uriV4;
+  }
+
+  @Override
+  Map<String, AttributeValue> createAttributes() {
+    // Check whether we are actually on ECS
+    if ((Strings.isNullOrEmpty(uriV3) && Strings.isNullOrEmpty(uriV4))) {
+      return ImmutableMap.of();
+    }
+
+    ImmutableMap.Builder<String, AttributeValue> resourceAttributes = ImmutableMap.builder();
+    try {
+      String hostName = InetAddress.getLocalHost().getHostName();
+      resourceAttributes.put(ResourceConstants.CONTAINER_NAME, stringAttributeValue(hostName));
+    } catch (UnknownHostException e) {
+      logger.log(Level.WARNING, "Could not get docker container name from hostname.", e);
+    }
+
+    String containerId = dockerHelper.getContainerId();
+    if (!Strings.isNullOrEmpty(containerId)) {
+      resourceAttributes.put(CONTAINER_ID, stringAttributeValue(containerId));
```

Comment:
According to the [specifications](https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/resource/semantic_conventions#container) there is not `container.id` field. Do you have a use case for this field?

Reply:
Ah nice eyes didn't notice this isn't semanticattributes.

@wangzlei shouldn't this be container name and the above hostname?

Reply:
We should definitely only be putting official semantic attributes into the Resource. And, the keys should be being kept in the `io.opentelemetry.sdk.resource.ResourceConstants` class.

Reply:
Yeah I think we'll look at proposing container.id as a semantic convention, it seems to be missing (name is usually human readable not the ID).

But wanted to confirm, you think there should only be official semantic attributes here? What about vendor-specific resource attributes? For example, AWS services have a log group ID used for correlation, which is sort of independent from other semantic values like the service name. I was sort of expecting we'd add this as a non-standard attribute on the resource at some point.

Reply:
Actually nevermind, I guess this log group I'm thinking of is x-ray specific, but not for general AWS users. So yeah doesn't belong here most likely :)

Reply:
Just realize that my previous reply was not sent out.
Refer to docker documentation [Container identification](https://docs.docker.com/engine/reference/run/#container-identification), Container's `full-id`, `short-id` and `name` are equivalent in docker CLI like 'docker info <id>'. `name` is usually human readable and can be changed by command ['docker rename'](https://docs.docker.com/engine/reference/commandline/rename/#examples) during runtime. 

I will cut a PR for adding `container.id` in semantic conversions. Thanks.